### PR TITLE
IFC-500 Please ASYNC linting rules

### LINKED ISSUE #1255: task: Activate ASYNC linter for ruff
### Component

API Server / GraphQL, Git Integration

### Task Description

We should activate the ASYNC linter for ruff and fix any code that breaks the rules.

```diff
diff --git a/pyproject.toml b/pyproject.toml
index afed7aaf..95c12688 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -308,6 +308,7 @@ disallow_untyped_defs = false
 [tool.ruff]

 select = [
+    "ASYNC",
     "C90",
     "DTZ",
     "E",
```

```
❯ ruff .
backend/infrahub/git/repository.py:1514:18: ASYNC101 Async functions should not call `open`, `time.sleep`, or `subprocess` methods
backend/infrahub/git/repository.py:1616:14: ASYNC101 Async functions should not call `open`, `time.sleep`, or `subprocess` methods
backend/tests/unit/api/test_60_storage.py:36:21: ASYNC101 Async functions should not call `open`, `time.sleep`, or `subprocess` methods
backend/tests/unit/git/conftest.py:201:10: ASYNC101 Async functions should not call `open`, `time.sleep`, or `subprocess` methods
backend/tests/unit/git/conftest.py:230:10: ASYNC101 Async functions should not call `open`, `time.sleep`, or `subprocess` methods
backend/tests/unit/git/conftest.py:261:10: ASYNC101 Async functions should not call `open`, `time.sleep`, or `subprocess` methods
backend/tests/unit/git/conftest.py:272:10: ASYNC101 Async functions should not call `open`, `time.sleep`, or `subprocess` methods
backend/tests/unit/git/conftest.py:315:14: ASYNC101 Async functions should not call `open`, `time.sleep`, or `subprocess` methods
backend/tests/unit/git/conftest.py:333:10: ASYNC101 Async functions should not call `open`, `time.sleep`, or `subprocess` methods
backend/tests/unit/git/test_git_repository.py:56:5: ASYNC101 Async functions should not call `open`, `time.sleep`, or `subprocess` methods
backend/tests/unit/git/test_git_repository.py:71:5: ASYNC101 Async functions should not call `open`, `time.sleep`, or `subprocess` methods
backend/tests/unit/git/test_git_repository.py:128:10: ASYNC101 Async functions should not call `open`, `time.sleep`, or `subprocess` methods
backend/tests/unit/git/test_git_repository.py:176:10: ASYNC101 Async functions should not call `open`, `time.sleep`, or `subprocess` methods
backend/tests/unit/git/test_git_repository.py:345:10: ASYNC101 Async functions should not call `open`, `time.sleep`, or `subprocess` methods
backend/tests/unit/git/test_git_repository.py:703:10: ASYNC101 Async functions should not call `open`, `time.sleep`, or `subprocess` methods
Found 15 errors.
```

Probably this would mostly mean to use [aiofiles](https://pypi.org/project/aiofiles/) instead of plain `open()`.

### PR DESCRIPTION
**Pull request by gmazoyer**
Fixes #1255

### CODE REVIEW

File: backend/infrahub/graphql/app.py [hunk 1]
```diff
@@ -185,7 +184,7 @@ async def _handle_http_request(
     ) -> JSONResponse:
         if request.app.state.response_delay:
             self.logger.info(f"Adding response delay of {request.app.state.response_delay} seconds")
-            time.sleep(request.app.state.response_delay)
```
Comment (ogenstad) [inline backend/infrahub/graphql/app.py:188]: This was something that @fatih-acar added to test some conditions where the app was completely blocked. I don't know if this section is still relevant to keep?

Comment (fatih-acar) [reply backend/infrahub/graphql/app.py:188]: Thanks for the heads up, yes this is still relevant, and I think we need to keep that blocking sleep. Can we just ignore the linting rule here?

Comment (gmazoyer) [reply backend/infrahub/graphql/app.py:188]: I kept it, just moved it to an `await asyncio.sleep()` one. Would it be OK like so?

Comment (fatih-acar) [reply backend/infrahub/graphql/app.py:188]: using asyncio.sleep will block the current async coroutine, not the whole app, that's why having time.sleep is useful here (we want to fully block the app to simulate execution lag/slowness, not sure if we will get the same behavior using asyncio.sleep)

**Review by ogenstad — APPROVED** [review]

**Pull request merged**